tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -53,9 +54,20 @@ public sealed class TransactionTestRunner : IJsonDrivenTestRunner, IDisposable
         #endregion
 
         // private fields
+        private string _collectionName = "test";
         private string _databaseName = "transaction-tests";
         private readonly List<IDisposable> _disposables = new List<IDisposable>();
-        private string _collectionName = "test";
+
+        public IMongoClient FailPointClient
+        {
+            get
+            {
+                var regularClient = DriverTestConfiguration.Client;
```

Comment:
This test suite requires using a connection string with `MultipleShardRouters` for sharded server

Reply:
This property doesn't seem to be named quite correctly. There are several places where it's called where we just randomly pick a writable server, and when called from those places it doesn't seem necessary.

The only relevant call is from from JsonDrivenTargetedFailPointTest. Why not limit the changes to that class?

Reply:
So, since the mentioned failpoint tests have `receiver="testRunner"`: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestFactory.cs#L85 I think it's expected that the data for them should(at least can) be provided from the test runner level